src/NServiceBus.RavenDB/Subscriptions/SubscriptionPersister.cs

```diff
@@ -102,22 +102,34 @@ public SubscriptionPersister(IDocumentStore store)
                 await session.SaveChangesAsync().ConfigureAwait(false);
             }
         }
+    
 
         public async Task<IEnumerable<Subscriber>> GetSubscriberAddressesForMessage(IEnumerable<MessageType> messageTypes, ContextBag context)
         {
-            var ids = messageTypes.Select(Subscription.FormatId).ToList();
-
             using (var suppressTransaction = new TransactionScope(TransactionScopeOption.Suppress, TransactionScopeAsyncFlowOption.Enabled))
             {
                 Subscriber[] subscribers;
                 using (var session = OpenAsyncSession())
                 {
+                    var collectionName = session.Advanced.DocumentStore.Conventions.FindTypeTagName(typeof(Subscription));
```

Comment:
I think it could be moved to the constructor. If your conventions are changing all through runtime you are in a serious mess. But feature init seems wrong because the feature shouldn't need to know this detail.